### DISCUSSION TITLE
Question on plugin configuration

### DISCUSSION THREAD

**Discussion by psylaux** [web-infra-dev/rsbuild#562] [Q&A] [answered]
Hello,,

I've been trying to get the [pluginAssetsRetry](https://rsbuild.dev/plugins/list/plugin-assets-retry.html) plugin set up but I come to learn that the plugins in rsbuild have a different structure than most other plugins. Most significantly, they dont have a class declaration, so when implementing I can't access the plugin.apply(compiler) function. I studied the docs, but the way it's written doesn't work for me. Can anyone help?

(ps: I also just posted [this question](https://github.com/web-infra-dev/rsbuild/discussions/561) related to the finding the config file, but figured this might be a separate question)

**Comment by chenjiahan** [comment, marked as answer]
Rsbuild plugins is just a function, it is different from webpack plugins.

You can read [this document](https://rsbuild.dev/plugins/dev/index.html) to learn about the Rsbuild plugins.
